spring/boot-autoconfigure/src/main/java/com/linecorp/armeria/spring/ArmeriaReactiveWebServerFactory.java

```diff
@@ -0,0 +1,281 @@
+import static com.linecorp.armeria.spring.MeterIdPrefixFunctionFactory.DEFAULT;
+import static java.util.Objects.requireNonNull;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.URL;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.TrustManagerFactory;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.boot.web.reactive.server.AbstractReactiveWebServerFactory;
+import org.springframework.boot.web.reactive.server.ReactiveWebServerFactory;
+import org.springframework.boot.web.server.Compression;
+import org.springframework.boot.web.server.Http2;
+import org.springframework.boot.web.server.Ssl;
+import org.springframework.boot.web.server.Ssl.ClientAuth;
+import org.springframework.boot.web.server.SslStoreProvider;
+import org.springframework.boot.web.server.WebServer;
+import org.springframework.core.io.buffer.DefaultDataBufferFactory;
+import org.springframework.http.server.reactive.HttpHandler;
+import org.springframework.util.ResourceUtils;
+
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpResponseWriter;
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.server.PathMapping;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.healthcheck.HealthChecker;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Metrics;
+import io.netty.handler.ssl.SslContextBuilder;
+
+/**
+ * {@link ReactiveWebServerFactory} which is used to create a new {@link ArmeriaWebServer}.
+ */
+public class ArmeriaReactiveWebServerFactory extends AbstractReactiveWebServerFactory {
+    private static final Logger logger = LoggerFactory.getLogger(ArmeriaReactiveWebServerFactory.class);
+
+    private final ConfigurableListableBeanFactory beanFactory;
+
+    /**
+     * Creates a new factory instance with the specified {@link ConfigurableListableBeanFactory}.
+     */
+    public ArmeriaReactiveWebServerFactory(ConfigurableListableBeanFactory beanFactory) {
+        this.beanFactory = requireNonNull(beanFactory, "beanFactory");
+    }
+
+    @Override
+    public WebServer getWebServer(HttpHandler httpHandler) {
+        final ServerBuilder sb = new ServerBuilder();
+
+        // TODO(hyangtack) Change factory and DataBuffer to Armeria-specific ones later if necessary.
+        final ArmeriaHttpHandlerAdapter handler =
+                new ArmeriaHttpHandlerAdapter(httpHandler, new DefaultDataBufferFactory());
```

Comment:
Curious if `DefaultDataBufferFactory` could be instantiated only once and reused.